proxy/http2/HPACK.cc

```diff
@@ -332,18 +332,41 @@ int64_t
 encode_string(uint8_t *buf_start, const uint8_t *buf_end, const char *value, size_t value_len)
 {
   uint8_t *p = buf_start;
+  bool use_huffman = true;
```

Comment:
According to [Compression Ratio](https://github.com/http2jp/hpack-test-case/wiki/Compression-Ratio) in [hpack-test-case](https://github.com/http2jp/hpack-test-case), the best compression rate is `0.31` but compression rate of naive-huffman imprementation (I guess this is similar to this PR) is `0.85`. So we have to back to here and implement smart algorithm.